core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -394,6 +394,27 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   }
 
   /**
+   * Request that the cluster manager kill the specified executor without adjusting the
+   * application resource requirements.
+   *
+   * The effect is that a new executor will be launched in place of the one killed by
+   * this request. This assumes the cluster manager will automatically and eventually
+   * fulfill all missing application resource requests.
+   *
+   * Return whether the request is received.
+   */
+  def killAndReplaceExecutor(executorId: String): Boolean = synchronized {
```

Comment:
OK, sounds good. I didn't do it initially because there isn't really a great name for the private method (we already have `doKillExecutors`). I'll figure something out.